src/plugins/index_pattern_editor/public/components/index_pattern_editor_flyout_content.tsx

```diff
@@ -210,7 +214,10 @@ const IndexPatternEditorFlyoutContentComponent = ({
         );
         timestampOptions = extractTimeFields(fields, requireTimestampField);
       }
-      if (isMounted.current) {
+      if (
+        isMounted.current &&
+        currentLoadingTimestampFieldsIdx === currentLoadingTimestampFieldsRef.current
```

Comment:
This is not required for the fix, but this is nice to have to remove race conditions